### ISSUE TITLE
[Accessibility] User is unable to navigate the Calendar using shortcut keys (Arrow keys, (Alt +) PageUp, PageDown, etc)

### ISSUE THREAD

**Issue by majornista** [adobe/coral-spectrum#147]
Transitioning away from Moment.js to DateTime breaks some keyboard functionality, because APIs for navigating by week or calculating the number of days in a month are missing in DateTime.

### Expected Behavior

1. User should be able to navigate Calendar grid by week using ArrowUp/ArrowDown, by month using PageUp/PageDown, and by year using Alt+PageUp/Alt+PageDown.
2. Calendar should account for the differences in length between months and navigate to the last day of the next or previous month when the current date does not appear in the next or previous month.
3. DateTime should account for Daylight Savings Time when calculating the difference between dates when one of the dates falls in Daylight Savings Time but the other date does not.

### Actual Behavior

1. User is unable to navigate Calendar grid by week using ArrowUp/ArrowDown, or by year using Alt+PageUp/Alt+PageDown.
2. Calendar does not account for the differences in length between month, so that when a user navigates from a date that does not appear in the next month focus is lost.
3. DateTime does not account for Daylight Savings Time when calculating the difference between dates when one of the dates falls in Daylight Savings Time but the other date does not, which causes the Calendar to add or lose a day when navigating between dates where one date is observing Daylight Savings Time while the other date does not.

### Reproduce Scenario (including but not limited to)

AEM: See: https://jira.corp.adobe.com/browse/CQ-4310946 and https://jira.corp.adobe.com/browse/CUI-7438

#### Steps to Reproduce

1. Open https://opensource.adobe.com/coral-spectrum/examples/#calendar
2. Navigate into the first Calendar instance using the Tab key.
3. Note that while you can navigate by day using ArrowLeft or ArrowRight keys, and by month using PageUp or PageDown, ArrowUp or ArrowDown do not navigate by week, and Alt + PageUp or Alt + PageDown do not navigate by year.
4. Navigate to Jan 31 of any year.
5. Try to navigate to February using PageDown key.
6. Notice that the cursor indicating the selected is lost when the Calendar transitions to February.

#### Browser name/version/os (ie Chrome Version 62.0.3202.94 (Official Build) (64-bit) MacOS)
Version 87.0.4280.88 (Official Build) (x86_64)

#### Coral Spectrum version 
v4.10.11
